identity_derive/src/impls/structs.rs

```diff
@@ -0,0 +1,662 @@
+                            Self { #(#fnames),* } => {
+                                #diff { #(#fields_into),* }
+                            }
+                        }
+                    }
+                }
+
+            }
+        }
+        // Tuple struct.
+        SVariant::Tuple => {
+            // get types and create markers for the positioned fields.
+            let field_typs: Vec<_> = fields.iter().map(|f| f.typ()).collect();
+            let field_max = field_typs.len();
+            let field_markers: Vec<Ident> = (0..field_max).map(|t| format_ident!("field_{}", t)).collect();
+
+            // get merge field logic.
+            let field_merge: Vec<TokenStream> = fields
+                .iter()
+                .map(|field| {
+                    let pos = field.position();
+                    if field.should_ignore() {
+                        quote! {
+                            self.#pos.clone(),
+                        }
+                    } else {
+                        quote! {
+                            if let Some(v) = diff.#pos {
+                                self.#pos.merge(v)
+                            } else {
+                                self.#pos.clone()
+                            },
+                        }
+                    }
+                })
+                .collect();
+
+            // get diff field logic.
+            let fields_diff: Vec<TokenStream> = fields
+                .iter()
+                .map(|field| {
+                    let pos = field.position();
+                    if field.should_ignore() {
+                        quote! {
+                            Option::None
+                        }
+                    } else if field.is_option() {
+                        quote! {
+                            if self.#pos != other.#pos && other.#pos != None {
+                                Some(self.#pos.diff(&other.#pos))
+                            } else {
+                                None
+                            },
+                        }
+                    } else {
+                        quote! {
+                            if self.#pos != other.#pos {
+                                Some(self.#pos.diff(&other.#pos))
+                            } else {
+                                None
+                            },
+                        }
+                    }
+                })
+                .collect();
+
+            // get from_diff field logic.
+            let fields_from: Vec<TokenStream> = fields
+                .iter()
+                .enumerate()
+                .map(|(idx, field)| {
+                    let marker = &field_markers[idx];
+                    let typ = field.typ();
+
+                    if field.should_ignore() {
+                        quote! { Default::default() }
+                    } else {
+                        quote! {
+                            <#typ>::from_diff(
```

Comment:
```
#marker.map(|v| <#typ>::from_diff(v)).unwrap_or_default()
```

What about converting fields like this? Avoids the `to/from` diff conversion for default values.